python/sparkdl/transformers/tf_tensor.py

```diff
@@ -66,65 +63,29 @@ def setParams(self, tfInputGraph=None, inputMapping=None, outputMapping=None, tf
-        # TODO: Cheap optimization: if all input tensors are of type float64, just do nothing here
-        injected_op_subgraph = tf.Graph()
-        # Maps names of input tensors in our original graph to outputs of the injected-op subgraph
-        input_map = {}
-        with injected_op_subgraph.as_default():
-            with tf.name_scope(self.SPARKDL_OP_SCOPE):
-                for _, orig_tensor_name in self.getInputMapping():
-                    orig_tensor = tfx.get_tensor(orig_tensor_name, user_graph)
-                    # Create placeholder with same shape as original input tensor, but that accepts
-                    # float64 input from Spark.
-                    spark_placeholder = tf.placeholder(tf.float64, shape=orig_tensor.shape,
-                                                       name=tfx.op_name(orig_tensor_name))
-                    # If the original tensor was of type float64, just pass through the Spark input
-                    if orig_tensor.dtype == tf.float64:
-                        input_map[orig_tensor_name] = spark_placeholder
-                    # Otherwise, cast the Spark input to the datatype of the original tensor
-                    else:
-                        input_map[orig_tensor_name] = tf.cast(spark_placeholder,
-                                                              dtype=orig_tensor.dtype)
-            tf.import_graph_def(graph_def=user_graph_def, input_map=input_map, name="")
-        return injected_op_subgraph.as_graph_def(add_shapes=True)
+            res = []
+            for _, tensor_name in self.getInputMapping():
+                placeholder_type = tfx.get_tensor(tensor_name, user_graph).dtype.as_datatype_enum
+                res.append(placeholder_type)
+        return res
 
     def _optimize_for_inference(self):
-        gin = self.getTFInputGraph()
-        # Inject cast ops to convert float64 input fed from Spark into the datatypes of the
-        # Graph's input nodes.
-        graphdef_with_casts = self._addCastOps(self.getTFInputGraph().graph_def)
-
+        graph_def = self.getTFInputGraph().graph_def
+        # Get data types of input placeholders
+        placeholder_types = self._get_placeholder_types(graph_def)
         # Strip away graph nodes not used in computing the tensors with the specified output names
-        input_names = [self._getSparkDlOpName(tnsr_name) for _, tnsr_name in self.getInputMapping()]
+        input_names = [tfx.op_name(tnsr_name) for _, tnsr_name in self.getInputMapping()]
         output_names = [tfx.op_name(tnsr_name) for tnsr_name, _ in self.getOutputMapping()]
-        opt_gdef = infr_opt.optimize_for_inference(graphdef_with_casts,
-                                                   input_names,
-                                                   output_names,
-                                                   tf.float64.as_datatype_enum)
-        return opt_gdef
+        return infr_opt.optimize_for_inference(graph_def,
```

Comment:
Replace each input tensor with a placeholder of the same type, instead of a `tf.float64` placeholder. See `optimize_for_inference` method signature: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/tools/optimize_for_inference_lib.py#L89